shell/components/nav/__tests__/Type.test.ts

```diff
@@ -1,167 +1,365 @@
-import { mount, RouterLinkStub } from '@vue/test-utils';
+import { shallowMount, RouterLinkStub, createLocalVue } from '@vue/test-utils';
 import Type from '@shell/components/nav/Type.vue';
+import { createChildRenderingRouterLinkStub } from '@shell/utils/unit-tests/ChildRenderingRouterLinkStub';
+import { TYPE_MODES } from '@shell/store/type-map';
+
+// Configuration
```

Comment:
Could we avoid having data outside the tests? It is very time-consuming to have all this list and try to figure out which test it is.

This is usually a code smell pointing to a convoluted structure of the unit tests.
Specifically if in your test you are addressing some common points, they must be shared in the same `describe()`.

I usually avoid comments like this and aim to deliver tests, but this one really took me too long, I can see already a lot of headaches in the migration phase.

Reply:
I think what you're suggesting is that we narrow the scope of the shared data, I think that's fair. I've gone ahead and I've done that. 

I extended a pattern which initially defined state outside of the tests https://github.com/rancher/dashboard/commit/ece9a33f2aaf94406bfb9b42255671bf5123f42d#diff-8c1a0fb993ec56591f5c80663d5ccf84607dee19981b5bc6ff54064fd23676edR8 I could see how you would see that as different but in my mind the pattern is consistent.

> I usually avoid comments like this and aim to deliver tests, but this one really took me too long, I can see already a lot of headaches in the migration phase.

Sorry if I'm misinterpreting this but given the context of the rest of this comment I think you're being impolite here. If you're frustrated with me and you'd like to discuss something, we can. If you're thinking I'm incompetent and don't want me working on vue3 or some other aspect of the app please let me and our managers know.

Reply:
No it's not about you. I have a personal issue about this which I am trying to address, so I apologize I have caused pressure on you and made you feel uncomfortable. I unfortunately sometimes do not realize it.
I hope you can understand and accept my apologies and also thanks for letting me know.
Just to be clear, it was not something aimed at you but at how I usually do reviews in general.

I will try to be more careful in the future.